protocol/src/fschacha20poly1305.rs

```diff
@@ -96,7 +94,7 @@ impl FSChaCha20Poly1305 {
     ///
     /// The 16-byte authentication tag.
     pub fn encrypt(&mut self, aad: &[u8], content: &mut [u8]) -> [u8; 16] {
-        let cipher = ChaCha20Poly1305::new(self.key, self.nonce());
+        let cipher = ChaCha20Poly1305::new(self.key, Nonce::new(self.nonce()));
```

Comment:
Maybe in future versions of `chacha` we can have the nonce implement `std::ops::Add`